### PR TITLE
Add weekly view

### PR DESCRIPTION
**Pull request by MajaMitreska**

### CODE REVIEW

**Comment by sardzoski** [comment]
117 files changed... 😵

File: PulseEcoSwiftUI/PulseEcoSwiftUI/CityMap/SensorDetails/SenDetView.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    var body: some View {
+        VStack {
+            RoundedRectangle(cornerRadius: CGFloat(5.0) / 2.0)
+                .frame(width: 40, height: 3.0)
+                .foregroundColor(Color(UIColor.systemGray2)).padding(.top, 10)
+            
+            
+            ChildSizeReader(size: self.$collapsedViewSize) {
+                
+                CollapsedView(viewModel: SensorDetailsVM(sensor: self.appVM.selectedSensor ?? SensorVM(),
```
Comment (sardzoski) [inline PulseEcoSwiftUI/PulseEcoSwiftUI/CityMap/SensorDetails/SenDetView.swift:29]: Create the VM in the VM 😊

File: PulseEcoSwiftUI/PulseEcoSwiftUI/CityMap/SensorDetails/SenDetView.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                                                         sensorsData: self.dataSource.sensorsData24h,
+                                                         selectedMeasure: self.dataSource.getCurrentMeasure(selectedMeasure: self.appVM.selectedMeasure)))
+                    .padding(.top, 5)
+                    .padding(.bottom, 40)
+                    .partialSheet(isPresented: self.$isSheetShown) {
+                        SensorDView(viewModel: ExpandedVM(sensorData24h: self.dataSource.sensorsData24h,
```
Comment (sardzoski) [inline PulseEcoSwiftUI/PulseEcoSwiftUI/CityMap/SensorDetails/SenDetView.swift:35]: Same here. Also you can extract this closure in a separate variable or method in this file and just reference it.

File: PulseEcoSwiftUI/PulseEcoSwiftUI/CityMap/SensorDetails/SenDetView.swift [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                                                          dailyAverages: self.dataSource.sensorsDailyAverageData))
+                            .padding(.bottom, 30)
+                }
+                
+            }
+        }.background(RoundedCorners(tl: 40, tr: 40, bl: 0, br: 0).fill(Color.white))
+            .offset(y: self.isExpanded ? UIHeight/2 - (self.expandedViewSize.height) + self.collapsedViewSize.height/2 : UIHeight/2 - (self.collapsedViewSize.height))
```
Comment (sardzoski) [inline PulseEcoSwiftUI/PulseEcoSwiftUI/CityMap/SensorDetails/SenDetView.swift:42]: Make this calculation in the view model

File: PulseEcoSwiftUI/PulseEcoSwiftUI/CityMap/SensorDetails/SenDetView.swift [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            .animation(.easeIn)
+            .transition(.slide)
+            .gesture(
+                DragGesture()
+                    .onEnded { value in
```
Comment (sardzoski) [inline PulseEcoSwiftUI/PulseEcoSwiftUI/CityMap/SensorDetails/SenDetView.swift:47]: Extract this closure in a separate property and remove the unneeded code.

File: PulseEcoSwiftUI/PulseEcoSwiftUI/CityMap/SensorDetails/SensorDView.swift [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+//  SensorDView.swift
+//  PulseEcoSwiftUI
+//
+//  Created by Maja Mitreska on 1/25/21.
+//  Copyright © 2021 Monika Dimitrova. All rights reserved.
+//
+
+import SwiftUI
+
+struct SensorDView: View {
```
Comment (sardzoski) [inline PulseEcoSwiftUI/PulseEcoSwiftUI/CityMap/SensorDetails/SensorDView.swift:11]: Same comment about the name, view models, data, commented out code, view models and logic referenced from the view model, busy body 😊

File: PulseEcoSwiftUI/PulseEcoSwiftUI/CityMap/SensorDetails/SensorDetailsView.swift [hunk 6]
```diff
@@ -1,38 +1,52 @@
 import SwiftUI
 
 struct SensorDetailsView: View {
```
Comment (sardzoski) [inline PulseEcoSwiftUI/PulseEcoSwiftUI/CityMap/SensorDetails/SensorDetailsView.swift:3]: Same thing here as well.

File: PulseEcoSwiftUI/PulseEcoSwiftUI/Constants.swift [hunk 7]
```diff
@@ -11,3 +11,6 @@ import SwiftUI
 
 let UIWidth = UIScreen.main.bounds.width
 let UIHeight = UIScreen.main.bounds.height
+let privacyPolicyLink = "https://www.netcetera.com/home/privacy-policy.html"
```
Comment (sardzoski) [inline PulseEcoSwiftUI/PulseEcoSwiftUI/Constants.swift:14]: Here is the config 😊

File: PulseEcoSwiftUI/PulseEcoSwiftUI/PartialViews/PartialSheetManager.swift [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+import Foundation
+import Combine
+import SwiftUI
+
+public enum PartialSheetState: String {
+    case hidden
+    case partial
+    case expanded
+}
+
```
Comment (sardzoski) [inline PulseEcoSwiftUI/PulseEcoSwiftUI/PartialViews/PartialSheetManager.swift:18]: Remove the extra lines

File: PulseEcoSwiftUI/PulseEcoSwiftUI/PartialViews/PartialSheetManager.swift [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+
+
+public class PartialSheetManager: ObservableObject {
```
Comment (sardzoski) [inline PulseEcoSwiftUI/PulseEcoSwiftUI/PartialViews/PartialSheetManager.swift:22]: This looks copied from StackOverflow :D

File: PulseEcoSwiftUI/PulseEcoSwiftUI/SceneDelegate.swift [hunk 10]
```diff
@@ -12,16 +12,15 @@ import SwiftUI
...
-    
     func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
         // Use this method to optionally configure and attach the UIWindow `window` to the provided UIWindowScene `scene`.
         // If using a storyboard, the `window` property will automatically be initialized and attached to the scene.
         // This delegate does not imply the connecting scene or session are new (see `application:configurationForConnectingSceneSession` instead).
         
         // Create the SwiftUI view that provides the window contents.
         //let contentView = ContentView()
-        let rootView = MainView()
+         let sheetManager: PartialSheetManager = PartialSheetManager()
```
Comment (sardzoski) [inline PulseEcoSwiftUI/PulseEcoSwiftUI/SceneDelegate.swift:22]: Wow, I hate this so much. But don't change this now

**Review by sardzoski — CHANGES_REQUESTED** [review]

**Comment by marko-mni** [comment]
Even tough there is a lot of work left for refactoring, we will merge this now to complete the internship goals and then I will refactor it.

**Pull request merged**